### PR TITLE
Update CI settings and use rubocop-jekyll

### PR DESCRIPTION
**Pull request by parkr**
Each commit describes the change!

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ inherit_gem:
   jekyll: .rubocop.yml
 
 AllCops:
-  TargetRubyVersion: 2.2
+  TargetRubyVersion: 2.4
```
[the remark this answers is not in the document]
Comment (parkr) [reply .rubocop.yml]: Yes! I want to teach @jekyllbot how to change these for us!

File: .travis.yml [hunk 2]
```diff
@@ -9,6 +9,10 @@ matrix:
   include:
     - rvm: 2.6
       env: JEKYLL_VERSION=3.7
+      os: linux
+    - rvm: 2.6
+      env: JEKYLL_VERSION=3.7
+      os: windows
```
Comment (DirtyF) [inline .travis.yml]: Ruby is not supported on Windows for now.

File: .rubocop.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashmaroli) [reply .rubocop.yml]: @parkr Is there a need to drop support for Ruby 2.3 syntax (apart from that its due for EOL in a couple of months)?
  - `rubocop-0.62.0` (and therefore `rubocop-jekyll`) still supports Ruby 2.3
  - Jekyll`-master` still supports Ruby 2.3 ..

[the remark this answers is not in the document]
Comment (DirtyF) [reply .rubocop.yml]: Note that [Bundler 2.0](https://github.com/bundler/bundler/releases/tag/v2.0.0) has already dropped support for Ruby 2.3.

[the remark this answers is not in the document]
Comment (ashmaroli) [reply .rubocop.yml]: @DirtyF  Bundler 2.0 dropped support for Ruby < 2.3 meaning versions older than 2.3.0

File: lib/jekyll-github-metadata/pages.rb [hunk 3]
```diff
@@ -32,13 +32,21 @@ def subdomain_isolation?
...
 
-        def enterprise?;  env == "enterprise" end
+        def enterprise?
+          env == "enterprise"
+        end
 
-        def development?; env == "development" end
+        def development?
+          env == "development"
+        end
```
[the remark this answers is not in the document]
Comment (parkr) [reply lib/jekyll-github-metadata/pages.rb]: Yeah, I'll be explicit here 😄

File: .travis.yml [hunk 4]
```diff
@@ -2,13 +2,14 @@ language: ruby
...
-  - 2.3
-env:
-  matrix:
-    - JEKYLL_VERSION=3.7
-    - JEKYLL_VERSION=3.8
+matrix:
+  include:
+    - rvm: 2.6
+      env: JEKYLL_VERSION=3.7.0
+      os: linux
```
Comment (ashmaroli) [inline .travis.yml]: Is this setting necessary since we're testing only on the *default OS*..?

**Comment by parkr** [comment]
@jekyllbot: merge +dev

**Pull request merged**